src/lightning/app/core/flow.py

```diff
@@ -488,6 +490,7 @@ def run(self):
         """
         if not user_key:
             frame = cast(FrameType, inspect.currentframe()).f_back
+            assert frame is not None
```

Comment:
If it has the potential to be None it may be better to raise a RuntimeError here rather than plain assert